front/components/vaults/VaultCreateWebsiteModal.tsx

```diff
@@ -1,32 +1,602 @@
+
+  const existingHeaders = webCrawlerConfiguration?.headers
+    ? Object.entries(webCrawlerConfiguration.headers).map(([key, value]) => {
+        return { key, value };
+      })
+    : [];
+  const [headers, setHeaders] = useState(existingHeaders);
+
+  const { mutateVaultDataSourceViews } = useVaultDataSourceViews({
+    workspaceId: owner.sId,
+    vaultId: vault.sId,
+    category: WEBSITE_CAT,
+  });
+
+  const frequencyDisplayText: Record<CrawlingFrequency, string> = {
+    never: "Never",
+    daily: "Every day",
+    weekly: "Every week",
+    monthly: "Every month",
+  };
+
+  const depthDisplayText: Record<DepthOption, string> = {
+    0: "0 level",
+    1: "1 level",
+    2: "2 levels",
+    3: "3 levels",
+    4: "4 levels",
+    5: "5 levels",
+  };
+
+  useEffect(() => {
+    if (isUrlValid(dataSourceUrl)) {
+      setDataSourceName(urlToDataSourceName(dataSourceUrl));
+    }
+  }, [dataSourceUrl]);
+
+  const validateForm = useCallback(() => {
+    let urlError = null;
+    let nameError = null;
+
+    // Validate URL
+    if (!isUrlValid(dataSourceUrl)) {
+      urlError =
+        "Please provide a valid URL (e.g. https://example.com or https://example.com/a/b/c)).";
+    }
+
+    // Validate Name
+    const nameExists = dataSources.some(
+      (d) => d.name === dataSourceName && d.id !== dataSourceOrView?.id
+    );
+    const dataSourceNameRes = isDataSourceNameValid(dataSourceName);
+    if (nameExists) {
+      nameError = "A Folder with the same name already exists";
+    } else if (!dataSourceName.length) {
+      nameError = "Please provide a name.";
+    } else if (dataSourceNameRes.isErr()) {
+      nameError = dataSourceNameRes.error;
+    }
+
+    setDataSourceUrlError(urlError);
+    setDataSourceNameError(nameError);
+    return !urlError && !nameError;
+  }, [dataSourceName, dataSources, dataSourceUrl, dataSourceOrView?.id]);
+
+  useEffect(() => {
+    if (isSubmitted) {
+      validateForm();
+    }
+  }, [dataSourceName, dataSourceUrl, isSubmitted, validateForm]);
+
+  const handleCreate = async () => {
+    setIsSubmitted(true);
+
+    if (!validateForm()) {
+      return;
+    }
+
+    setIsSaving(true);
+    if (webCrawlerConfiguration === null) {
+      const sanitizedDataSourceUrl = dataSourceUrl.trim();
+      const res = await fetch(
+        `/api/w/${owner.sId}/vaults/${vault.sId}/data_sources/managed`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            provider: "webcrawler",
+            connectionId: "none",
+            name: dataSourceName,
+            configuration: {
+              url: sanitizedDataSourceUrl,
+              maxPageToCrawl: maxPages || WEBCRAWLER_MAX_PAGES,
+              depth: maxDepth,
+              crawlMode: crawlMode,
+              crawlFrequency: selectedCrawlFrequency,
+              headers: headers.reduce(
```

Comment:
Would be better to extract this logic above so we can use proper typings over `satisfies`